server/src/test/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapperTests.java

```diff
@@ -176,6 +176,20 @@ public void testNullField() throws Exception {
         assertEquals(0, fields.length);
     }
 
+    public void testBlankFieldName() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(this::minimalMapping));
+        ParsedDocument parsedDoc = mapper.parse(source(b -> b.startObject("field").field("", "value").endObject()));
+        IndexableField[] fields = parsedDoc.rootDoc().getFields("field");
+        assertEquals(2, fields.length);
+    }
+
+    public void testDotOnlyFieldName() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(this::minimalMapping));
+        ParsedDocument parsedDoc = mapper.parse(source(b -> b.startObject("field").field("...", "value").endObject()));
+        IndexableField[] fields = parsedDoc.rootDoc().getFields("field");
+        assertEquals(2, fields.length);
+    }
```

Comment:
It would be good to have an additional test with a couple more ordinary fields with dots in their names at the same level as the flattened field, just to make sure that dots are expanded for them.

Reply:
Thanks! I haven't tried it, but I suspect that your additional test would succeed even if you forgot to reset the withinLeafObject flag :) that's because at the lucene level, the fact that we map objects makes no difference.
To be 100% sure, I would check that the dynamic mappings generated by the additional fields have objects. a should be an object mapper that holds b as an objectMapper which holds c as a field mapper.

Reply:
Done.